distribution/kpkginstall/runtest.sh

```diff
@@ -304,65 +313,73 @@ function copr_prepare()
 }
 
 function rpm_install()
 {
-  echo "Extracting kernel version from ${KPKG_URL}"
+  print_info "Extracting kernel version from ${KPKG_URL}"
   get_kpkg_ver
   if [ -z "${KVER}" ]; then
     abort_recipe "Failed to extract kernel version from the package" FAIL
   else
-    echo "Kernel version is ${KVER}"
+    print_success "Kernel version is ${KVER}"
   fi
 
   # Ensure that the debug kernel is selected as the default kernel in
   # /boot/grub2/grubenv.
   if [[ "${KPKG_VAR_DEBUG:no}" == "yes" ]]; then
     echo "Adjusting settings in /etc/sysconfig/kernel to set debug as default"
-    echo "UPDATEDEFAULT=yes" | tee /etc/sysconfig/kernel
-    echo "DEFAULTKERNEL=kernel-debug" | tee -a /etc/sysconfig/kernel
-    echo "DEFAULTDEBUG=yes" | tee -a /etc/sysconfig/kernel
+    echo "UPDATEDEFAULT=yes" > /etc/sysconfig/kernel
+    echo "DEFAULTKERNEL=kernel-debug" >> /etc/sysconfig/kernel
+    echo "DEFAULTDEBUG=yes" >> /etc/sysconfig/kernel
+    print_success "Updated /etc/sysconfig/kernel to set debug kernels as default"
   fi
 
   # download & install kernel, or report result
   download_install_package "${PACKAGE_NAME}-$KVER" "kernel"
 
 
-  if $YUM install -y "${PACKAGE_NAME}-devel-${KVER}" 2>&1; then
-    echo "✅ Installed ${PACKAGE_NAME}-devel-${KVER} successfully"
+  if $YUM install -y "${PACKAGE_NAME}-devel-${KVER}" > /dev/null; then
+    print_success "Installed ${PACKAGE_NAME}-devel-${KVER} successfully"
   else
-    echo "❌ No package kernel-devel-${KVER} found, skipping!"
-    echo "⚠️  Note that some tests might require the package and can fail!"
+    print_warning "No package kernel-devel-${KVER} found, skipping!"
+    print_warning "Note that some tests might require the package and can fail!"
   fi
-  if $YUM install -y "${PACKAGE_NAME}-headers-${KVER}" 2>&1; then
-    echo "❌ No package kernel-headers-${KVER} found, skipping!"
-    echo "⚠️  Note that some tests might require the package and can fail!"
+  if $YUM install -y "${PACKAGE_NAME}-headers-${KVER}" > /dev/null; then
+    print_success "Installed ${PACKAGE_NAME}-headers-${KVER} successfully"
+  else
+    print_warning "No package kernel-headers-${KVER} found, skipping!"
+    print_warning "Note that some tests might require the package and can fail!"
   fi
 
-  # The package was renamed (and temporarily aliased) in Fedora/RHEL
+  # The package was renamed (and temporarily aliased) in Fedora/RHEL"
   if $YUM search kernel-firmware | grep "^kernel-firmware\.noarch" ; then
-      $YUM install -y kernel-firmware 2>&1
+    FIRMWARE_PKG=kernel-firmware
   else
-      $YUM install -y linux-firmware 2>&1
+    FIRMWARE_PKG=linux-firmware
   fi
+  print_info "Installing kernel firmware package"
+  $YUM install -y $FIRMWARE_PKG > /dev/null
+  print_success "Kernel firmware package installed"
 
   # Workaround for BZ 1698363
   if [[ "${ARCH}" == s390x ]] ; then
-    grubby --set-default /boot/"${KVER}" && zipl
+    grubby --set-default /boot/"${KVER}" > /dev/null
+    zipl > /dev/null
```

Comment:
Hm... was it on purpose to execute `zipl` regardless of the success of `grubby`?

Reply:
Oops, I didn't meant to do that.